bokehjs/src/lib/models/glyphs/annulus.ts

```diff
@@ -84,12 +84,25 @@ export class AnnulusView extends XYGlyphView {
   protected _hit_point(geometry: PointGeometry): Selection {
     const {sx, sy} = geometry
     const x = this.renderer.xscale.invert(sx)
-    const x0 = x - this.max_outer_radius
-    const x1 = x + this.max_outer_radius
-
     const y = this.renderer.yscale.invert(sy)
-    const y0 = y - this.max_outer_radius
-    const y1 = y + this.max_outer_radius
+
+    let x0: number, y0: number
+    let x1: number, y1: number
+    if (this.model.properties.outer_radius.units == "data") {
+      x0 = x - this.max_outer_radius;
+      x1 = x + this.max_outer_radius;
+
+      y0 = y - this.max_outer_radius;
+      y1 = y + this.max_outer_radius;
+    } else {
+      const sx0 = sx - this.max_outer_radius;
+      const sx1 = sx + this.max_outer_radius;
+      [x0, x1] = this.renderer.xscale.r_invert(sx0, sx1);
+
+      const sy0 = sy - this.max_outer_radius;
+      const sy1 = sy + this.max_outer_radius;
```

Comment:
isn't `this.max_outer_radius` in data units? I would expect this distance would need to be converted to screen units before adding or subtracting to `sx` or `sy`

Reply:
Or better, to use a max of `souter_radius` since that's the value that is actually used in the call to `ctx.arc`

Reply:
Or alternatively: for annulus we could just skip the spatial indexing altogether and make things simpler. i.e. just iterate over and test all the annuli always, instead of trying to get a list of candidates from the index first. I doubt users are trying sho show thousands of annulus values

Reply:
this.max_outer_radius seems to be in the same units as what is specified. This follows a similar approach to annular wedge, https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/glyphs/annular_wedge.ts#L96.

Reply:
It appears to be being added to "x" (data units) in the first block and "sx" (screen units) in the second. If so, those can't both be correct